lib/Dialect/ESI/runtime/python/esiaccel/types.py

```diff
@@ -260,6 +262,8 @@ def is_valid(self, obj) -> Tuple[bool, Optional[str]]:
 
   def serialize(self, obj) -> bytearray:
     ret = bytearray()
+    if not isinstance(obj, dict):
```

Comment:
Is `is_valid` called before this? If not, it should be. Problem is that it should either be called at the top of the serialize call or be called here and not descend into the inner objects. Or is it better to require the user code to check and put it somewhere in the documentation?